analysis/common_variables.py

```diff
@@ -173,347 +173,552 @@ def generate_common_variables(index_date_variable,end_date_variable):
-    #     parkinson_icd10,
-    #     returning="date_of_death",
-    #     on_or_after=f"{index_date_variable}",
-    #     match_only_underlying_cause=True,
-    #     date_format="YYYY-MM-DD",
-    #     return_expectations={
-    #         "date": {"earliest": "index_date", "latest" : "today"}, #Ask for index date
-    #         "rate": "uniform",
-    #         "incidence": 0.1,
-    #     },
+    # #
+    # out_date_parkinson_antipsychotic_prescription=patients.minimum_of(
+    #     "tmp_out_date_parkinson_antipsychotic_prescription_bnf",
     # ),
-    #         # Prescriptions (BNF)
 
-    #     ## Restless leg syndrome (Condition on the clinical pathway to outcome of interest)
-    #         # Primary
-    #         # HES
-    # tmp_out_date_restless_leg_syndrome_hes=patients.admitted_to_hospital(
-    #     returning="date_admitted",
-    #     with_these_diagnoses=restless_leg_syndrome_icd10,
-    #     on_or_after=f"{index_date_variable}",
-    #     date_format="YYYY-MM-DD",
-    #     find_first_match_in_period=True,
-    #     return_expectations={
-    #         "date": {"earliest": "index_date", "latest" : "today"}, #Ask for index date
-    #         "rate": "uniform",
-    #         "incidence": 0.1,
-    #     },
-    # ),        
-    #         # ONS
-    # tmp_out_date_restless_leg_syndrome_death=patients.with_these_codes_on_death_certificate(
-    #     restless_leg_syndrome_icd10,
-    #     returning="date_of_death",
-    #     on_or_after=f"{index_date_variable}",
-    #     match_only_underlying_cause=True,
-    #     date_format="YYYY-MM-DD",
-    #     return_expectations={
-    #         "date": {"earliest": "index_date", "latest" : "today"}, #Ask for index date
-    #         "rate": "uniform",
-    #         "incidence": 0.1,
-    #     },
-    # ),
-    #         # Prescriptions (BNF)
+################################
+# Neurodegenerative Covariates # 
+################################
 
-    #     ## REM sleep disorder (Condition on the clinical pathway to outcome of interest)
-    #         # Primary
-    #         # HES
-    # tmp_out_date_rem_sleep_syndrome_hes=patients.admitted_to_hospital(
-    #     returning="date_admitted",
-    #     with_these_diagnoses=rem_sleep_syndrome_icd10,
-    #     on_or_after=f"{index_date_variable}",
-    #     date_format="YYYY-MM-DD",
-    #     find_first_match_in_period=True,
-    #     return_expectations={
-    #         "date": {"earliest": "index_date", "latest" : "today"}, #Ask for index date
-    #         "rate": "uniform",
-    #         "incidence": 0.1,
-    #     },
-    # ),
-    #         # ONS
-    # tmp_out_date_rem_sleep_syndrome_death=patients.with_these_codes_on_death_certificate(
-    #     rem_sleep_syndrome_icd10,
+    # Hypercholesterolaemia 
+    # Primary
+    tmp_cov_bin_hypercholesterolaemia_snomed=patients.with_these_clinical_events(
+        hypercholesterolaemia_snomed,
+        returning="date",
+        on_or_after=f"{index_date_variable}",
```

Comment:
for covariates we use on_or_before don't we ? or is it on purpose for these particular ones ?

Reply:
and we use - 1 day. 
for cholesterol I've been told to use between=[f"{index_date_variable}- 5years", f"{index_date_variable} -1 day"]
I think you need to check which one to use

Reply:
Thanks, I changed. I am using 1680 days (5 years) for precission.